sqlmesh/dbt/basemodel.py

```diff
@@ -293,25 +299,54 @@ def __init__(self, context: DbtContext, config: BMC):
             dialect=context.engine_adapter.dialect if context.engine_adapter else "",
         )
 
+        self.jinja_env = self.context.jinja_macros.build_environment(**self._jinja_globals)
+
     @property
     def enriched_config(self) -> BMC:
         if self._rendered_sql is None:
+            self._enriched_config = self._update_with_sql_config(self._enriched_config)
             self.render()
             self._enriched_config._dependencies = self._enriched_config._dependencies.union(
                 self._captured_dependencies
             )
         return self._enriched_config
 
+    @property
+    def parsed_jinja(self) -> nodes.Template:
+        if self._parsed_jinja is None:
+            self._parsed_jinja = self.jinja_env.parse(self.config.all_sql)
+        return self._parsed_jinja
+
     def render(self) -> str:
         if self._rendered_sql is None:
-            registry = self.context.jinja_macros
-            self._rendered_sql = (
-                registry.build_environment(**self._jinja_globals)
-                .from_string(self.config.all_sql)
-                .render()
-            )
+            self._rendered_sql = self.jinja_env.from_string(self.parsed_jinja).render()
         return self._rendered_sql
 
+    def _update_with_sql_config(self, config: BMC) -> BMC:
+        def _extract_value(node: t.Any) -> t.Any:
+            if not isinstance(node, nodes.Node):
+                return node
+            if isinstance(node, nodes.Const):
+                return _extract_value(node.value)
+            if isinstance(node, nodes.List):
+                return [_extract_value(val) for val in node.items]
+            if isinstance(node, nodes.Dict):
+                return {_extract_value(pair.key): _extract_value(pair.value) for pair in node.items}
+            if isinstance(node, nodes.Tuple):
+                return tuple(_extract_value(val) for val in node.items)
+            if isinstance(node, nodes.Call):
```

Comment:
Previously I pointed out a call as an example but what I said is true for other kind of expressions too. For example:
```
config(
  attr=1+1
)
```
I assume we still need to evaluate these. Should we consider doing:
```
self.jinja_env.from_string(nodes.Template([nodes.Output([node])])).render()
```
as a fallback instead of throwing an exception?

Btw, sorry, I should've clarified this from the start.